### PR TITLE
Add tree sequence metadata and schema to low-level python

### PR DESCRIPTION
**Pull request by benjeffery**
Low-level python part of #532, stacked on #642 

- [x] TableCollection properties
- [x] TreeSequence properties

### CODE REVIEW

File: python/_tskitmodule.c [hunk 1]
```diff
@@ -6073,6 +6073,86 @@ TableCollection_get_file_uuid(TableCollection *self, void *closure)
     return Py_BuildValue("s", self->tables->file_uuid);
 }
 
+static PyObject *
+TableCollection_get_metadata(TableCollection *self, void *closure)
+{
+    PyObject *ret = NULL;
+
+    /* Py_BuildValue returns Py_None for zero length, we would rather
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:6081]: We're doing this a few times, so we should make a function for it, i.e., ``get_metadata_bytes`` or something.

File: python/_tskitmodule.c [hunk 2]
```diff
@@ -6073,6 +6073,86 @@ TableCollection_get_file_uuid(TableCollection *self, void *closure)
...
+
+static int
+TableCollection_set_metadata(TableCollection *self, PyObject *arg, void *closure)
+{
+    int ret = -1;
+    int err;
+    char *metadata;
+    Py_ssize_t metadata_length;
+
+    if (arg == NULL) {
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:6099]: Similarly, we're probably doing the same thing here for a bunch of tables/table collections. How about a function 
```c
if (parse_metadata_arg(arg, &metadata, &metadata_length) != 0){
    goto out;
}
At least then we're not rewriting the same error strings.
```

File: python/_tskitmodule.c [hunk 3]
```diff
@@ -6073,6 +6073,86 @@ TableCollection_get_file_uuid(TableCollection *self, void *closure)
...
+
+static int
+TableCollection_set_metadata_schema(TableCollection *self, PyObject *arg, void *closure)
+{
+    int ret = -1;
+    int err;
+    const char *metadata_schema;
+    Py_ssize_t metadata_schema_length;
+
+    if (arg == NULL) {
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:6135]: Same here - we're repeating the same code a good few times.

File: python/_tskitmodule.c (hunk 1 above)
Comment (benjeffery) [reply python/_tskitmodule.c]: Replaced with `PyBytes_FromStringAndSize`.

File: python/_tskitmodule.c (hunk 2 above)
Comment (benjeffery) [reply python/_tskitmodule.c:6073]: I've factored this out for schemas, but as table collection metadata has a different type and message have left it as is.

File: python/_tskitmodule.c (hunk 3 above)
Comment (benjeffery) [reply python/_tskitmodule.c]: As above.

File: python/_tskitmodule.c (hunk 2 above)
Comment (jeromekelleher) [reply python/_tskitmodule.c:6073]: Nice, LGTM

File: python/_tskitmodule.c (hunk 1 above)
Comment (jeromekelleher) [reply python/_tskitmodule.c]: Even better.

**Comment by benjeffery** [comment]
@jeromekelleher Ready for review

```
(env) benj@treebeard:~/projects/tskit/python$ python stress_lowlevel.py -m lowlevel
iter	tests	err	fail	skip	RSS	min	max	max@iter
309	174	0	0	0	80320	79976	80320	1
```

File: python/_tskitmodule.c [hunk 4]
```diff
@@ -6676,6 +6724,18 @@ TreeSequence_get_site(TreeSequence *self, PyObject *args)
     return ret;
 }
 
+static PyObject * TreeSequence_get_metadata(TreeSequence * self) {
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:6727]: Formatting problem, 
```
static PyObject *
TreeSequence_get_metadata...
{
```

File: python/_tskitmodule.c [hunk 5]
```diff
@@ -6676,6 +6724,18 @@ TreeSequence_get_site(TreeSequence *self, PyObject *args)
...
+    return PyBytes_FromStringAndSize(
+        self->tree_sequence->tables->metadata, 
+        self->tree_sequence->tables->metadata_length);
+}
+
+static PyObject * TreeSequence_get_metadata_schema(TreeSequence * self) {
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:6733]: Ditto

File: python/_tskitmodule.c (hunk 4 above)
Comment (benjeffery) [reply python/_tskitmodule.c]: Fixed!

File: python/_tskitmodule.c (hunk 5 above)
Comment (benjeffery) [reply python/_tskitmodule.c]: Fixed!

**Comment by benjeffery** [comment]
@mergify refresh

**Comment by benjeffery** [comment]
travis is AWOL, attempting to kick it into gear

**Pull request merged**